pkg/reporter/sarif_reporter.go

```diff
@@ -0,0 +1,127 @@
+package reporter
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+type SARIFReporter struct {
+	outputDest string
+}
+
+type SARIFLog struct {
+	Version string `json:"version"`
+	Schema  string `json:"$schema"`
+	Runs    []runs `json:"runs"`
+}
+
+type runs struct {
+	Tool    tool     `json:"tool"`
+	Results []result `json:"results"`
+}
+
+type tool struct {
+	Driver driver `json:"driver"`
+}
+
+type driver struct {
+	Name    string `json:"name"`
+	InfoURI string `json:"informationUri"`
+	Version string `json:"version"`
+}
+
+type result struct {
+	Kind      string     `json:"kind"`
+	Level     string     `json:"level"`
+	Message   message    `json:"message"`
+	Locations []location `json:"locations"`
+}
+
+type message struct {
+	Text string `json:"text"`
+}
+
+type location struct {
+	PhysicalLocation physicalLocation `json:"physicalLocation"`
+}
+
+type physicalLocation struct {
+	ArtifactLocation artifactLocation `json:"artifactLocation"`
+}
+
+type artifactLocation struct {
+	URI string `json:"uri"`
+}
+
+func NewSARIFReporter(outputDest string) *SARIFReporter {
+	return &SARIFReporter{
+		outputDest: outputDest,
+	}
+}
+
+func createSARIFReport(reports []Report) (*SARIFLog, error) {
+	var log SARIFLog
+
+	n := len(reports)
+
+	log.Version = "2.1.0"
```

Comment:
Recommend making these constants  and moving them outside this function

Reply:
done